server/src/main/java/io/crate/auth/JWTAuthenticationMethod.java

```diff
@@ -102,7 +107,10 @@ public JWTAuthenticationMethod(Roles roles,
                 audience = jwtProperties.aud() != null ? jwtProperties.aud() : audience;
             }
 
-            JwkProvider jwkProvider = urlToJwkProvider.apply(decodedJWT.getIssuer());
+            String jwtIssuer = decodedJWT.getIssuer();
+            Duration jwtCacheDuration = connProperties == null ? null : connProperties.jwtKeyExpiration();
```

Comment:
`ConnectionProperties` can be null only in test scenarios. An alternative would be to define ConnectionProperties in all tests.